### PR TITLE
Log to application insights

### LINKED ISSUE #382: Send log messages to Application Insight
**Describe the new feature you would like to see**
* All log messages that are logged to console should also be sent to Application Insight and stored.

Include possibility to log based on log level.

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #382 
We might need to consider linking ISAR to its keyvault for better management of the connection string.
Currently there needs to exist an environment variable called `APPLICATIONINSIGHTS_CONNECTION_STRING` containing the connection string for the correct Application Insights in Azure.

Currently logging to:
https://portal.azure.com/#@StatoilSRM.onmicrosoft.com/resource/subscriptions/c389567b-2dd0-41fa-a5da-d86b81f80bda/resourceGroups/IsarDev/providers/Microsoft.Insights/components/IsarDevAI/overview

Go to Logs and query `traces` to see log messages.
Otherwise, you can find the request (API -request/responses) logging in the overview.

### CODE REVIEW

File: src/isar/apis/api.py [hunk 1]
```diff
@@ -5,17 +5,27 @@
...
 import click
 import uvicorn
-from fastapi import FastAPI, Security
+from fastapi import FastAPI, Security, Request
 from fastapi.middleware.cors import CORSMiddleware
 from fastapi.routing import APIRouter
 from injector import inject
 from pydantic import AnyHttpUrl
 
+# Opencensus imports
```
Comment (Christdej) [inline src/isar/apis/api.py:14]: Unnecessary comment in my opinion

File: src/isar/config/log.py [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
 
-    for loggers in log_config["loggers"].keys():
-        logging.getLogger(loggers).addHandler(log_handler)
-        logging.getLogger(loggers).setLevel(log_levels[loggers])
-    logging.getLogger().addHandler(log_handler)
+def configure_azure_handler(log_config: dict) -> logging.Handler:
+    # Automatically gets connection string from env variable 'APPLICATIONINSIGHTS_CONNECTION_STRING'
+    handler = AzureLogHandler()
+    handler.setLevel(log_config["root"]["level"])
+    return handler
```
Comment (Christdej) [inline src/isar/config/log.py:45]: Add `APPLICATIONINSIGHTS_CONNECTION_STRING` to settings.py with description and a default?

**Review by Christdej — CHANGES_REQUESTED** [review]

File: src/isar/apis/api.py [hunk 3]
```diff
@@ -80,6 +90,26 @@ def _create_app(self) -> FastAPI:
 
         app.include_router(router=self._create_info_router())
 
+        @app.middleware("http")
+        async def middlewareOpencensus(request: Request, call_next):
+            tracer = Tracer(
+                exporter=AzureExporter(),
```
Comment (Christdej) [inline src/isar/apis/api.py:96]:
```suggestion
                exporter=AzureExporter(connection_string=settings.APPLICATIONINSIGHTS_CONNECTION_STRING),
```

File: src/isar/apis/api.py (hunk 1 above)
Comment (GodVenn) [reply src/isar/apis/api.py:14]: Agree

File: src/isar/config/log.py (hunk 2 above)
Comment (GodVenn) [reply src/isar/config/log.py:45]: Good idea

File: src/isar/apis/api.py (hunk 3 above)
Comment (GodVenn) [reply src/isar/apis/api.py:96]: This one also gets it automatically (like the log handler), should i set both explicitly or neither?

File: src/isar/config/log.py (hunk 2 above)
Comment (GodVenn) [reply src/isar/config/log.py:45]: Although, it is a secret, should it still be added?

File: src/isar/apis/api.py (hunk 3 above)
Comment (GodVenn) [reply src/isar/apis/api.py:96]: Will set both explicitly

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**